### PR TITLE
feat: Add Plasma Icons support

### PR DESCRIPTION
**Pull request by ItachiSan**
Add the relevant Plasma plugin.

### CODE REVIEW

File: yin_yang/plugins/icons.py [hunk 1]
```diff
@@ -64,3 +67,35 @@ def available_themes(self) -> dict:
...
+
+        for icon_theme in theme_directories:
+            if not path.isdir(icon_theme):
+                continue
+
+            for icon_theme_folder in scandir(icon_theme):
+                if not all(
+                    (
+                        icon_theme_folder.is_dir(),
+                        path.isfile(icon_theme_folder.path + "/index.theme"),
```
Comment (l0drex) [inline yin_yang/plugins/icons.py:89]: If I remember correctly, you can do `(icon_theme_folder / 'index.theme').is_file()`. I think that looks a bit nicer.

File: yin_yang/plugins/icons.py [hunk 2]
```diff
@@ -64,3 +67,35 @@ def available_themes(self) -> dict:
...
+                        path.isfile(icon_theme_folder.path + "/icon-theme.cache"),
+                    )
+                ):
+                    continue
+
+                theme_parser = configparser.ConfigParser(strict=False)
+                theme_parser.read(icon_theme_folder.path + "/index.theme")
```
Comment (l0drex) [inline yin_yang/plugins/icons.py:96]: Similar to the above: `read(icon_theme_folder / 'index.theme')`

File: yin_yang/plugins/icons.py (hunk 1 above)
Comment (ItachiSan) [reply yin_yang/plugins/icons.py:89]: Seems not:
Screenshot_20240403_212745

File: yin_yang/plugins/icons.py (hunk 2 above)
Comment (ItachiSan) [reply yin_yang/plugins/icons.py:96]: See above

File: yin_yang/plugins/icons.py (hunk 1 above)
Comment (ItachiSan) [reply yin_yang/plugins/icons.py:89]: We are using `scandir` from `os`, not `pathlib`

**Comment by ItachiSan** [comment]
@l0drex did you see my replies to your comments? :smile:

**Comment by l0drex** [comment]
Yes, looks good then :+1:

**Review by l0drex — APPROVED** [review]

**Comment by chase9** [comment]
FYI, this does not work on Fedora since the path for plasma-changeicons is `/usr/libexec/plasma-changeicons` instead of `/usr/lib/plasma-changeicons`.

**Pull request merged**